dmutils/csrf.py

```diff
@@ -27,6 +27,9 @@ def check_valid_csrf():
         request.form.get(TOKEN, None),
         request.headers.get(REACT_HEADER_NAME, None)
     ]
+    json = request.get_json()
+    if json:
+        tokens_received.append(json.get(TOKEN, None))
```

Comment:
Because POST etc requests from the frontend will typically contain the request data as JSON encoded, this is needed to get the CSRF token from the request body.